astropy/units/tests/test_units.py

```diff
@@ -651,13 +651,13 @@ def test_compare_with_none():
     # Ensure that equality comparisons with `None` work, and don't
     # raise exceptions.  We are deliberately not using `is None` here
     # because that doesn't trigger the bug.  See #3108.
-    assert not (u.m == None)
-    assert u.m != None
+    assert not (u.m is None)
```

Comment:
The comment right above this one says, "We are deliberately not using `is None` here because that doesn't trigger the bug. See #3108."